include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -410,10 +410,21 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
-        : __my_it_(__it), __my_unary_func_(__unary_func)
+    //only enable this constructor if both template types are default constructible
+    template <typename _Iter_ = _Iter, typename _UnaryFunc_ = _UnaryFunc,
```

Comment:
Should we really have here `typename _Iter_ = _Iter, typename _UnaryFunc_ = _UnaryFunc,` ?

Reply:
We can use different names for the types, but yes I believe we need these rather than just relying on the structure's templates.  If we make these not dependent on template parameters specific to the function, it will cause problems for the structure as a whole I believe.

Reply:
ok